src/main/frontend/worker/commands.cljs

```diff
@@ -0,0 +1,179 @@
+(ns frontend.worker.commands
+  "Invoke commands based on user settings"
+  (:require [datascript.core :as d]
+            [logseq.db.frontend.property.type :as db-property-type]
+            [cljs-time.core :as t]
+            [cljs-time.coerce :as tc]
+            [logseq.db.frontend.property :as db-property]
+            [logseq.outliner.pipeline :as outliner-pipeline]
+            [frontend.worker.handler.page.db-based.page :as worker-db-page]
+            [logseq.common.util.date-time :as date-time-util]))
+
+;; TODO: allow users to add command or configure it through #Command (which parent should be #Code)
```

Comment:
I plan to store commands using blocks; when the DB changes, we'll know which commands should be executed, e.g., commands for changed blocks, and commands for changed blocks' tags.  
We probably still need to separate it by graphs if we want to cache the commands to avoid querying from the db.